frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -159,22 +163,33 @@ const sortByRevision = (
   return _.toArray(replicators).sort(compare);
 };
 
-const getAnnotation = (obj: K8sResourceKind, annotation: string): string => {
-  return _.get(obj, ['metadata', 'annotations', annotation]);
+const getAnnotation = (obj: K8sResourceCommon, annotation: string): string => {
+  return obj?.metadata?.annotations?.[annotation];
 };
 
-const getDeploymentRevision = (obj: K8sResourceKind): number => {
+export const getDeploymentRevision = (obj: K8sResourceCommon): number => {
   const revision = getAnnotation(obj, DEPLOYMENT_REVISION_ANNOTATION);
   return revision && parseInt(revision, 10);
 };
 
-const getDeploymentConfigVersion = (obj: K8sResourceKind): number => {
+export const getDeploymentConfigVersion = (obj: K8sResourceCommon): number => {
   const version = getAnnotation(obj, DEPLOYMENT_CONFIG_LATEST_VERSION_ANNOTATION);
   return version && parseInt(version, 10);
 };
 
-const getDeploymentConfigName = (obj: K8sResourceKind): string => {
-  return _.get(obj, 'metadata.ownerReferences[0].name', null);
+export const getOwnerNameByKind = (obj: K8sResourceCommon, kind: K8sKind): string => {
+  const reference = referenceForModel(kind);
+  if (!reference) {
+    return null;
+  }
+  const groupVersionKind = getGroupVersionKind(reference);
+  if (!groupVersionKind) {
+    return null;
+  }
+  const apiVersion = `${groupVersionKind[0]}/${groupVersionKind[1]}`;
+  return obj?.metadata?.ownerReferences?.find(
+    (ref) => ref.kind === kind.kind && ref.apiVersion === apiVersion,
```

Comment:
I think we should ignore the actual version if possible since the same resource can have multiple versions. Maybe something like this (not tested)

```suggestion
    (ref) => ref.kind === kind.kind && ((!kind.apiGroup && ref.apiVersion === 'v1') || ref.apiVersion?.startsWith(`${kind.apiGroup}/`)),
```